### PR TITLE
WIP setup: add flask-shell-ipython dependency

### PR DESCRIPTION
**Pull request by jmartinm**
* Allows `$ inspirehep shell` to use Ipython instead of default
  Python shell.

Signed-off-by: Javier Martin Montull <javier.martin.montull@cern.ch>

### CODE REVIEW

**Comment by jacquerie** [comment]
This merge conflict really makes me think that we need a commit that applyes some rules for the order of stuff in `setup.py`.

File: setup.py [hunk 1]
```diff
@@ -31,7 +31,13 @@
 
 install_requires = [
     'amqp>=1.4.9,<2.0',
+    'Flask>=0.11.1',
+    'Flask-Admin>=1.4.2',
+    'Flask-Breadcrumbs>=0.3.0',
+    'Flask-Script>=2.0.5',
+    'Flask-CLI>=0.4.0',
```
Comment (jirikuncar) [inline setup.py]: do not use Flask-CLI with Flask>=0.11.1

Comment (jirikuncar) [inline setup.py]: do not use Flask-CLI with Flask>=0.11.1

Comment (jacquerie) [inline setup.py]: It's required by a lot of other packages anyway, `invenio-search` included: https://github.com/inveniosoftware/invenio-search/blob/0fc8f760fa3f5abb47e02eaa70735778cdc9fc3f/setup.py#L71

Comment (jirikuncar) [inline setup.py]: > It's required by a lot of other packages anyway

Yes, but we are going to remove this dependency and your overlay should not depend on it directly. The same applies to most of Flask* dependencies in your setup.py. Pin only top-level dependencies and use `pip-tools` to compile your full production requirements.

Comment (jacquerie) [inline setup.py]: Yes, in fact I resubmitted this patch without a dependency on Flask-CLI: https://github.com/inspirehep/inspire-next/pull/1319#issue-165122304. Our `setup.py` is without doubt full of scar tissue.

**Pull request closed without merging**